### PR TITLE
Fix #2120

### LINKED ISSUE #2120: Flaky test reporter should not create duplicated issues for subtests
From discussion: https://knative.slack.com/archives/CA1DTGZ2N/p1590494902288600

Example:
* https://github.com/knative/eventing-contrib/issues/1262
* https://github.com/knative/eventing-contrib/issues/1261

The two issues are about the same flaky test, but the bot open two issues

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

**What this PR does, why we need it**:
Fix duplicated issues for flaky test reporter

**Which issue(s) this PR fixes**:
Fixes #2120

/assign @chaodaiG

### CODE REVIEW

File: tools/flaky-test-reporter/result.go [hunk 1]
```diff
@@ -158,6 +158,29 @@ func addSuiteToRepoData(suite *junit.TestSuite, buildID int, rd *RepoData) {
 	}
 }
 
+// https://github.com/knative/test-infra/issues/2120
+func filterOutParentTests(originalCases []junit.TestCase) []junit.TestCase {
+	parents := make(map[string]interface{})
```
Comment (chaodaiG) [inline tools/flaky-test-reporter/result.go:163]: There is a nicer `set` to use :)
https://godoc.org/k8s.io/apimachinery/pkg/util/sets

**Comment by chaodaiG** [comment]
/lgtm
/approve

/hold
please work with our oncall @albertomilan updating the `flaky-test-reporter` image based on instruction here: https://github.com/knative/test-infra/tree/master/images

**Comment by chizhg** [comment]
I have updated the image.

**Comment by chizhg** [comment]
I have updated the image.

**Pull request merged**